database/mysql/mysql.go

```diff
@@ -54,20 +54,22 @@ type Mysql struct {
 	config *Config
 }
 
-// instance must have `multiStatements` set to true
-func WithInstance(instance *sql.DB, config *Config) (database.Driver, error) {
+// connection instance must have `multiStatements` set to true
+func WithConnection(conn *sql.Conn, config *Config) (*Mysql, error) {
```

Comment:
Thoughts on making this context aware? [Eventually](https://github.com/golang-migrate/migrate/issues/14), we want most DB driver methods to be context aware. I'm thinking that new methods should be context aware.

Also, unfortunately, these functions that create DB drivers currently return interfaces instead of concrete times. I'm thinking returning a concrete type as you're doing now is a cleaner pattern. Can you add a type check? e.g. `var _ database.Driver = (*Mysql)(nil) // explicit compile time type check`

Reply:
If it's consistent across all drivers, then it would be a good thing. But it seems out of scope for this PR.

> Can you add a type check?
Sorry, I don't understand what you are asking. Where do you want this check?

Reply:
> If it's consistent across all drivers, then it would be a good thing. But it seems out of scope for this PR.

Oh yeah, making the db drivers consistent is not in the scope of the PR. But new code should fit within the future plans/vision if possible. In this case, we should add a `ctx context.Context` parameter to `WithConnection()`

> > Can you add a type check?
> 
> Sorry, I don't understand what you are asking. Where do you want this check?

Add `var _ database.Driver = (*Mysql)(nil) // explicit compile time type check` as a package level ignored variable.

Reply:
OK, I understood your point.

Reply:
Done (both of it).